tests/unit/utils/sparqlhandler/test_csvw_inspector.py

```diff
@@ -55,49 +54,52 @@ def test_get_primary_catalog_metadata_key_error():
-    object created from it, in this case to correctly detect a QbDataset csvw type from its csvw_state.
+    object created from it, in this case to correctly detect a QbDataset csvw type from its csvw_inspector.
     """
     csvw_metadata_json_path = (
         _test_case_base_dir
         / "pivoted-single-measure-dataset"
         / "qb-id-10004.csv-metadata.json"
     )
-    csvw_rdf_manager = get_csvw_rdf_manager(csvw_metadata_json_path)
+    csvw_rdf_manager = CsvwRdfManager(csvw_metadata_json_path)
 
-    csvw_type = csvw_rdf_manager.csvw_state.csvw_type
+    csvw_type = csvw_rdf_manager.csvw_inspector.csvw_type
     assert csvw_type == CSVWType.QbDataSet
 
 
-def test_csvw_type_key_error():
+def test_detect_csvw_type_code_list():
     """
-    Test to ensure that when a conjunctive graph that is not recognised by the
-    csvw_type() function as a code list or a qb data set returns a type error.
+    Tests the detection of the csvw_type of an input metadata json file that has a CsvwRdfManager
+    object created from it, in this case to correctly detect a CodeList csvw type from its csvw_inspector.
     """
-
-    rdf_graph = ConjunctiveGraph()
-
-    csvw_inspector = CsvWState(rdf_graph, Path("does-not-exist.csv-metadata.json"))
-
-    with pytest.raises(TypeError) as exception:
-        assert csvw_inspector.csvw_type()
-
-    assert (
-        "The input metadata is invalid as it is not a data cube or a code list."
-        in str(exception.value)
+    csvw_metadata_json_path = (
+        _test_case_base_dir
+        / "pivoted-single-measure-dataset"
+        / "some-dimension.csv-metadata.json"
     )
+    csvw_rdf_manager = CsvwRdfManager(csvw_metadata_json_path)
 
+    csvw_type = csvw_rdf_manager.csvw_inspector.csvw_type
+    assert csvw_type == CSVWType.CodeList
 
-def test_detect_csvw_type_code_list():
+
+def test_get_table_info_for_csv_url():
     """
-    Tests the detection of the csvw_type of an input metadata json file that has a CsvwRdfManager
-    object created from it, in this case to correctly detect a CodeList csvw type from its csvw_state.
+    Ensures that the correct table schema properties are returned for the given code list.
     """
     csvw_metadata_json_path = (
         _test_case_base_dir
         / "pivoted-single-measure-dataset"
         / "some-dimension.csv-metadata.json"
     )
-    csvw_rdf_manager = get_csvw_rdf_manager(csvw_metadata_json_path)
+    csvw_rdf_manager = CsvwRdfManager(csvw_metadata_json_path)
 
-    csvw_type = csvw_rdf_manager.csvw_state.csvw_type
-    assert csvw_type == CSVWType.CodeList
+    csvw_inspector: CsvWInspector = CsvWInspector(
+        csvw_rdf_manager.rdf_graph, csvw_metadata_json_path
+    )
+
+    result = csvw_inspector.get_table_info_for_csv_url("some-dimension.csv")
+
+    assert result.about_url == "some-dimension.csv#{+uri_identifier}"
+    assert result.csv_url == "some-dimension.csv"
+    assert result.primary_key_col_names == ["uri_identifier"]
```

Comment:
Can you add a test for a CSV-W with multiple columns in the primary key?

If you use a cube (with locally defined code lists) then you'll be able to test a CSV-W that has multiple tables in it as well - which is another test I'd like to see.

Reply:
Multiple columns test added. Do we need to add a "get_all_tables" function before we add the second test here?

Reply:
I don't think so, you can just access the private property.

Reply:
Done